File: flutter_app/lib/models/plant.dart
class Plant {
  final int id;
  final String scientificName;
  final String commonName;
  final String family;
  final String nativeRegion;
  final String conservationStatus;
  final String ecologicalImportance;
  final String description;
  final String threats;
  final String conservationActions;
  final String habitat;
  final String identificationFeatures;
  final String? imageUrl;
  final String? plantnetSpeciesName;

  Plant({
    required this.id,
    required this.scientificName,
    required this.commonName,
    required this.family,
    required this.nativeRegion,
    required this.conservationStatus,
    required this.ecologicalImportance,
    required this.description,
    required this.threats,
    required this.conservationActions,
    required this.habitat,
    required this.identificationFeatures,
    this.imageUrl,
    this.plantnetSpeciesName,
  });

  factory Plant.fromJson(Map<String, dynamic> json) {
    return Plant(
      id: json['id'] as int,
      scientificName: json['scientific_name'] ?? '',
      commonName: json['common_name'] ?? '',
      family: json['family'] ?? '',
      nativeRegion: json['native_region'] ?? '',
      conservationStatus: json['conservation_status'] ?? '',
      ecologicalImportance: json['ecological_importance'] ?? '',
      description: json['description'] ?? '',
      threats: json['threats'] ?? '',
      conservationActions: json['conservation_actions'] ?? '',
      habitat: json['habitat'] ?? '',
      identificationFeatures: json['identification_features'] ?? '',
      imageUrl: json['image_url'],
      plantnetSpeciesName: json['plantnet_species_name'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'scientific_name': scientificName,
      'common_name': commonName,
      'family': family,
      'native_region': nativeRegion,
      'conservation_status': conservationStatus,
      'ecological_importance': ecologicalImportance,
      'description': description,
      'threats': threats,
      'conservation_actions': conservationActions,
      'habitat': habitat,
      'identification_features': identificationFeatures,
      'image_url': imageUrl,
      'plantnet_species_name': plantnetSpeciesName,
    };
  }
}

File: backend/app/models/plant.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, Index
from app.database import Base


class Plant(Base):
    __tablename__ = "plants"

    id = Column(Integer, primary_key=True, index=True)
    scientific_name = Column(String(255), nullable=False, unique=False, index=True)
    common_name = Column(String(255), nullable=False, index=True)
    family = Column(String(255), nullable=False)
    native_region = Column(String(255), nullable=False)
    conservation_status = Column(String(100), nullable=False)
    ecological_importance = Column(Text, nullable=False)
    description = Column(Text, nullable=False)
    threats = Column(Text, nullable=False)
    conservation_actions = Column(Text, nullable=False)
    habitat = Column(String(255), nullable=False)
    identification_features = Column(Text, nullable=False)
    image_url = Column(String(500), nullable=True)
    plantnet_species_name = Column(String(255), nullable=True, index=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    __table_args__ = (
        Index("idx_plant_scic_name", "scientific_name"),
        Index("idx_plant_common_name", "common_name"),
        Index("idx_plant_plantnet_name", "plantnet_species_name"),
    )
